Generate API-reference also

arve0 opened the pull request:
Docstrings are already nice formatted, lets make use of it!

arve0 commented:
Generated a demo build: http://dask-doc-demo.readthedocs.org/en/apidocs/index.html

arve0 commented:
The readthedocs build fails on subpackages, but works locally, so thats probably just the setup over at readthedocs.

shoyer commented:
In my experience, you get much more useable API docs if you do some manual organization and explicitly call out each function/class you want to document. Otherwise you end up with too much nesting.

arve0 commented:
I agree. I do not have enough knowledge about the dask API to manually organize it, but this gives a good starting point. Also, scaffolding the API-docs once in a while is a good idea, its easy to miss putting a new submodule in the docs.

mrocklin commented:
Agree with both.  I'll see if I can find some time this or next week to curate this a bit.

arve0 commented:
Removed the unnecessary `sphinxcontrib-napoleon` which is included in `sphinx>=1.3`.

Also removed one level of nesting. API-reference now goes directly to the dask-package. There is at most 2 steps of nesting now, API-reference->subpackage.

mrocklin commented on dask/utils.py:
```diff
@@ -139,15 +139,14 @@ def textblock(file, start, stop, compression=None):
-    >> textblock(f, 1, 10)
-    '456\n789\n'
+    >>> textblock(f, 1, 10)
+    '456\\n789\\n'
```
This was actually intentional.  Some doctesters don't parse results with endlines well.

mrocklin commented:
I'm going to close this in favor of solutions like #236 .  Thanks for spurring this on!
